test/test_tensorclass_nofuture.py

```diff
@@ -354,6 +354,47 @@ class MyDataNested:
     ):
         data[:2] = data3[:2]
 
+    # regression test PR #203
+    # We should be able to set tensors items with MemmapTensors and viceversa
+    @tensorclass
+    class MyDataMemMap1:
+        x: torch.Tensor
+        y: MemmapTensor
+
+    @tensorclass
+    class MyDataMemMap2:
+        x: MemmapTensor
+        y: torch.Tensor
+
+    data1 = MyDataMemMap1(
+        x=torch.zeros(3, 4, 5),
+        y=MemmapTensor.from_tensor(torch.zeros(3, 4, 5)),
+        batch_size=[3, 4],
+    )
+    data2_wrong = MyDataMemMap2(
+        x=MemmapTensor.from_tensor(torch.ones(3, 4, 5)),
+        y=torch.ones(3, 4, 5),
+        batch_size=[3, 4],
+    )
+    with pytest.raises(ValueError, match="__setitem__ is only allowed for same-class assignement"):
+        data1[:2] = data2_wrong[:2]
+    with pytest.raises(ValueError, match="__setitem__ is only allowed for same-class assignement"):
+        data2_wrong[2:] = data1[2:]
+        
+    data2 = MyDataMemMap1(
+        x=MemmapTensor.from_tensor(torch.ones(3, 4, 5)),
+        y=torch.ones(3, 4, 5),
+        batch_size=[3, 4],
+    )    
+    data1[:2] = data2[:2]
+    assert (data1[:2] == 1).all()
```

Comment:
this currently returns False. I think we should treat scalars differently and perform a broadcast @vmoens, @tcbegley, @sreevasthav wdyt?

Reply:
Yep
Happy to implement some broadcasting for `TensorDIct` first with float, integers, tensors and memmap-tensors in `__setitem__`.

Reply:
This actually works as of today
```python
>>> from tensordict import TensorDict
td = TensorDict>>> td = TensorDict({"a": [1, 1, 1]}, [])
>>> td==1
TensorDict(
    fields={
        a: Tensor(shape=torch.Size([3]), device=cpu, dtype=torch.bool, is_shared=False)},
    batch_size=torch.Size([]),
    device=None,
    is_shared=False)
>>> (td==1)["a"]
tensor([True, True, True])
```
but this does not 
```
>>> td==torch.zeros(3)
False
```
I can make a quick patch

Reply:
I also think it makes sense to compare two tensorclass of a different type, but I might get this wrong.
Our type checks work with `isinstance`, hence we'll get some weird behaviours if one class inherits from another on the lhs or rhs (ie `a==b` is not `b==a`)

Reply:
> I also think it makes sense to compare two tensorclass of a different type, but I might get this wrong. Our type checks work with `isinstance`, hence we'll get some weird behaviours if one class inherits from another on the lhs or rhs (ie `a==b` is not `b==a`)

we can relax the constraint to be just a tensorclass with all keys equal